borsh/tests/test_zero_sized_types.rs

```diff
@@ -0,0 +1,142 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg(feature = "derive")]
+
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+#[cfg(not(feature = "std"))]
+use alloc::{format, vec, vec::Vec};
+
+#[cfg(feature = "std")]
+use std::collections::{BTreeMap, BTreeSet, HashMap, HashSet, LinkedList, VecDeque};
+
+#[cfg(not(feature = "std"))]
+use alloc::collections::{BTreeMap, BTreeSet, LinkedList, VecDeque};
+#[cfg(feature = "hashbrown")]
+use hashbrown::{HashMap, HashSet};
+
+use borsh::from_slice;
+use borsh::to_vec;
+use borsh::BorshDeserialize;
+use borsh::BorshSerialize;
+
+use borsh::error::ERROR_ZST_FORBIDDEN;
+#[derive(BorshDeserialize, BorshSerialize, PartialEq, Debug, Eq, PartialOrd, Ord, Hash)]
+struct A();
+
+#[test]
+fn test_deserialize_vec_of_zst() {
+    let v = [0u8, 0u8, 0u8, 64u8];
+    let res = from_slice::<Vec<A>>(&v);
+    assert_eq!(format!("{}", res.unwrap_err()), ERROR_ZST_FORBIDDEN);
```

Comment:
This reads a bit better:

```suggestion
    assert_eq!(res.unwrap_err().to_string(), ERROR_ZST_FORBIDDEN);
```